backend/views/syson-diagram-general-view/src/main/java/org/eclipse/syson/diagram/general/view/services/GeneralViewNodeToolSectionSwitch.java

```diff
@@ -60,83 +62,130 @@ public class GeneralViewNodeToolSectionSwitch extends SysmlEClassSwitch<Void> {
     public GeneralViewNodeToolSectionSwitch(NodeDescription nodeDescription, List<NodeDescription> allNodeDescriptions) {
         this.viewBuilderHelper = new ViewBuilders();
         this.diagramBuilderHelper = new DiagramBuilders();
-        this.nodeToolSections = new ArrayList<>();
         this.nodeDescription = Objects.requireNonNull(nodeDescription);
         this.allNodeDescriptions = Objects.requireNonNull(allNodeDescriptions);
     }
 
-    public List<NodeToolSection> getNodeToolSections() {
-        return this.nodeToolSections;
+    private NodeToolSection buildCreateSection(NodeTool... nodeTools) {
```

Comment:
Could you move this new `buildCreateSection` method after all `caseXXX` methods please?

Reply:
Done.